packages/engine/paima-sm/src/delegate-wallet.ts

```diff
@@ -0,0 +1,302 @@
+import { PaimaParser } from '@paima/concise';
+import { ENV, doLog } from '@paima/utils';
+import { CryptoManager } from '@paima/crypto';
+import Web3 from 'web3';
+import type { PoolClient } from 'pg';
+import { error } from 'console';
+import type { IGetAddressFromAddressResult } from '@paima/db';
+import {
+  deleteAddress,
+  deleteDelegationTo,
+  deleteDelegationsFrom,
+  getAddressFromAddress,
+  getAddressFromId,
+  getDelegation,
+  getDelegationsFrom,
+  getDelegationsTo,
+  newAddress,
+  newDelegation,
+  updateAddress,
+  updateDelegateFrom,
+} from '@paima/db';
+
+type ParsedDelegateWalletCommand =
+  | {
+      command: 'delegate';
+      args: { from: string; to: string; from_signature: string; to_signature: string };
+    }
+  | {
+      command: 'migrate';
+      args: { from: string; to: string; from_signature: string; to_signature: string };
+    }
```

Comment:
Initially I had `delegate` and `migrate` as different things. I think it still makes sense conceptually to separate these, but in practice it may be a little tricky so if you want to only have `delegate` and no `migrate` that's fine I think.

Reply:
They are implemented, but I can remove them if you don't want them.